src/components/channel-effects/ChannelEffectSubscriber.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PubSub from 'pubsub-js';
+
+import { Checkbox } from 'antd';
+import ChannelSlider from './ChannelSlider';
+import PopoverColor from '../sets/PopoverColor';
+import { VIEWER_PALETTE } from '../utils';
+
+import TitleInfo from '../TitleInfo';
+import {
+  SLIDERS_CHANGE, RASTER_ADD, COLORS_CHANGE, CHANNEL_TOGGLE,
+} from '../../events';
+
+export default class ChannelEffectSubscriber extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { colorValues: {}, channelsOn: {} };
+    this.setSliderValue = this.setSliderValue.bind(this);
+    this.setColorValue = this.setColorValue.bind(this);
+    this.toggleChannel = this.toggleChannel.bind(this);
+    this.componentWillUnmount = this.componentWillUnmount.bind(this);
+  }
+
+  componentWillMount() {
+    this.rasterAddToken = PubSub.subscribe(RASTER_ADD, this.rasterAddSubscriber.bind(this));
+  }
+
+  componentDidMount() {
+    const { onReady } = this.props;
+    onReady();
+  }
+
+  componentWillUnmount() {
+    PubSub.unsubscribe(this.rasterAddToken);
+  }
+
+  rasterAddSubscriber(msg, sliderData) {
+    Object.keys(sliderData.channels).sort().forEach((channel, i) => {
+      this.setColorValue({ channel, color: VIEWER_PALETTE[i] });
+      this.toggleChannel(channel);
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  setSliderValue(sliderValue) {
+    PubSub.publish(SLIDERS_CHANGE, sliderValue);
+  }
+
+  setColorValue({ channel, color }) {
+    const colorValue = {};
```

Comment:
```suggestion
    const colorValue = {[channel]: color};
```
or even do it inline below.